### DISCUSSION TITLE
difference between SendingTime and received time

### DISCUSSION THREAD

**Discussion by pe-ggy-lee** [quickfix-j/quickfixj#401] [Q&A]
I am using the quickfix version ==> 1.6.3. thesedays we want to improve our recieve data speedy performance, we found that the server sending time and the quickfix receiving time differed by 40 milliseconds,how do i can improve this issue.

**Comment by chrjohn** [comment]
Hi @pe-ggy-lee ,

that is quite an old version that you got there. Did you try with a newer version?
However, I think this is no general problem but a problem related to either your setup or your application.
Are the server and your machine time synchronised?
Is the first message of the day (i.e. the Logon) also 40ms late?

Cheers,
Chris.

**Reply by pe-ggy-lee** [reply]
can i contract with you? i'm using new version but tag 52 sendingtime format have millisecond, how can i process this??

**Reply by pe-ggy-lee** [reply]
server  and client are in same machine

**Reply by chrjohn** [reply]
OK, then I think the problem is application or machine-specific. In one of our applications we usually receive and forward messages from remote machines inside 1ms.
Are you able to provide a log file?

**Comment by pe-ggy-lee** [comment]
Hello

my log there~

Server log

09:01:21.053 [MSG] DeQ 8=FIX.4.3 9=340 35=UP10 49=SM03 56=GM960$ 34=506
52=20210601-01:01:21 80001=03 80002=08 80003=10 80014=         80024=090121
80004=0000 80005=0001 80006=04980 11=00260 81029=0004246 81039=00010.6500
81040=00005000 81041=00010.7500 81028=00005000 81048=090121028
81032=0004001 81042=00010.8500 81043=00005000 81044=00010.8500
81031=00005000 81049=090102146 10=255



Quickfixlog

20210601-01:01:21.101: 8=FIX.4.3 9=325 35=UP10 49=emgMsgSvr 56=960T1111
34=8214 52=20210601-01:01:21 80001=03 80002=08 80003=10 80014=
80024=090121 80004=0000 11=00260 81029=0004246 81039=00010.6500
81040=00005000 81041=00010.7500 81028=00005000 81048=090121028
81032=0004001 81042=00010.8500 81043=00005000 81044=00010.8500
81031=00005000 81049=090102146 10=241



Client log [log time is GMT+8]

2021-06-01
09:01:21.100|SocketConnectorIoProcessor-0.0hread|quickfix.mina.message.FIXMessageDecoder.parseMessage(FIXMessageDecoder.java:226)|
parsed message: pos=348,lim=604,rem=256,offset=348,state=4 8=FIX.4.3 9=325
35=UP10 49=emgMsgSvr 56=960T1111 34=8214 52=20210601-01:01:21 80001=03
80002=08 80003=10 80014=         80024=090121 80004=0000 11=00260
81029=0004246 81039=00010.6500 81040=00005000 81041=00010.7500
81028=00005000 81048=090121028 81032=0004001 81042=00010.8500
81043=00005000 81044=00010.8500 81031=00005000 81049=090102146 10=241

2021-06-01
09:01:21.101|SocketConnectorIoProcessor-0.0hread|quickfix.mina.message.FIXMessageDecoder.parseMessage(FIXMessageDecoder.java:157)|
detected header: pos=348,lim=604,rem=256,offset=348,state=1

2021-06-01
09:01:21.101|SocketConnectorIoProcessor-0.0hread|quickfix.mina.message.FIXMessageDecoder.parseMessage(FIXMessageDecoder.java:176)|
body length = 233: pos=348,lim=604,rem=256,offset=364,state=3

2021-06-01
09:01:21.101|SocketConnectorIoProcessor-0.0hread|quickfix.mina.message.FIXMessageDecoder.parseMessage(FIXMessageDecoder.java:200)|
message body found: pos=348,lim=604,rem=256,offset=597,state=4

2021-06-01
09:01:21.101|SocketConnectorIoProcessor-0.0hread|quickfix.mina.message.FIXMessageDecoder.parseMessage(FIXMessageDecoder.java:207)|
found checksum: pos=348,lim=604,rem=256,offset=597,state=4

2021-06-01 09:01:21.101|QF/J Session dispatcher:
FIX.4.3:960T1111->emgMsgSvrhread|com.fubon.rstsfix.RstsFIX$OTCFIX.fromApp(RstsFIX.java:1269)|
QuickfixQueueSize:1  IN  <- 8=FIX.4.3 9=325 35=UP10 34=8214 49=emgMsgSvr
52=20210601-01:01:21 56=960T1111 11=00260 80001=03 80002=08 80003=10
80004=0000 80014=         80024=090121 81028=00005000 81029=0004246
81031=00005000 81032=0004001 81039=00010.6500 81040=00005000
81041=00010.7500 81042=00010.8500 81043=00005000 81044=00010.8500
81048=090121028 81049=090102146 10=241


Christoph John ***@***.***> 於 2021年6月10日 週四 下午6:32寫道：

**Comment by pe-ggy-lee** [comment]
I have been looking for this problem for several days, and there is no
problem sending the information, thank you for your help

Peggy Lee ***@***.***> 於 2021年6月10日 週四 下午6:59寫道：

**Reply by chrjohn** [reply]
That's kind of hard to follow. The time difference is between the sending from the server (server log), and the client (quickfix log and client log)?
Maybe you can improve the speed already by turning off that DEBUG logging for each message?

Again, can you check the time difference on the Logon message after an application restart?

**Comment by pe-ggy-lee** [comment]
Does this mean that writing files affects speed? ? I have been closed to
DEBUG, but the situation is still the same....

Christoph John ***@***.***> 於 2021年6月10日 週四 下午7:11寫道：

**Comment by chrjohn** [comment]
What do you mean? Every operation that you do will influence performance eventually. 
Please answer all of my other questions from before. Especially the one about the Logon.
Thanks

**Comment by pe-ggy-lee** [comment]
sorry...

that's kind of hard to follow. The time difference is between the sending
from the server (server log), and the client (quickfix log and client log)?

==>yes,
Maybe you can improve the speed already by turning off that DEBUG logging
for each message?

==> Tried, but the same result

Again, can you check the time difference on the Logon message after an
application restart?

==>is it mean like this message?
[image: image.png]



Christoph John ***@***.***> 於 2021年6月10日 週四 下午7:57寫道：

**Reply by chrjohn** [reply]
> Again, can you check the time difference on the Logon message after an
> application restart?
> 
>> ==>is it mean like this message?

There is no image shown. Can you maybe copy/paste the text directly from the log file?
